scripts/write_nz_building_outline.py

```diff
@@ -0,0 +1,209 @@
+import json
+import sys
+import warnings
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, List
+
+import click
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+import pygeos
+from numpy.typing import NDArray
+
+GEOPARQUET_VERSION = "0.3.0"
+AVAILABLE_COMPRESSIONS = ["NONE", "SNAPPY", "GZIP", "BROTLI", "LZ4", "ZSTD"]
+
+PygeosGeometryArray = NDArray[pygeos.Geometry]
+
+
+class PathType(click.Path):
+    """A Click path argument that returns a pathlib Path, not a string"""
+
+    def convert(self, value, param, ctx):
+        return Path(super().convert(value, param, ctx))
+
+
+class GeometryType(int, Enum):
+    """Pygeos (GEOS) geometry type mapping
+    From https://pygeos.readthedocs.io/en/latest/geometry.html?highlight=type#pygeos.geometry.get_type_id
+    """
+
+    Missing = -1
+    Point = 0
+    LineString = 1
+    LinearRing = 2
+    Polygon = 3
+    MultiPoint = 4
+    MultiLinestring = 5
+    MultiPolygon = 6
+    GeometryCollection = 7
+
+
+def parse_to_pygeos(df: gpd.GeoDataFrame) -> Dict[str, PygeosGeometryArray]:
+    """Parse to pygeos geometry array
+
+    This is split out from _create_metadata so that we don't have to create the pygeos
+    array twice: once for converting to wkb and another time for metadata handling.
+    """
+    geometry_columns: Dict[str, PygeosGeometryArray] = {}
+    for col in df.columns[df.dtypes == "geometry"]:
+        pygeos_geoms = pygeos.from_shapely(df[col])
```

Comment:
This step is not really needed, because if pygeos is installed, geopandas will automatically use pygeos under the hood (so this step is basically converting the pygeos geometries to shapely (because `pygeos.from_shapely` converts the Series to a numpy object array, which will output shapely geometries) and then back to pygeos geometries)

Reply:
You can get the numpy array with pygeos geometries directly as `df[col].array.data`